src/Toast.js

```diff
@@ -56,76 +56,79 @@ const defaultProps = {
   transition: Fade,
 };
 
-const Toast = ({
-  bsPrefix,
-  className,
-  children,
-  transition: Transition,
-  show,
-  animation,
-  delay,
-  autohide,
-  onClose,
-  innerRef,
-  ...props
-}) => {
-  const delayRef = useRef(delay);
-  const onCloseRef = useRef(onClose);
-
-  // We use refs for these, because we don't want to restart the autohide
-  // timer in case these values change.
-  delayRef.current = delay;
-  onCloseRef.current = onClose;
-
-  useEffect(() => {
-    if (!(autohide && show)) {
-      return undefined;
-    }
-
-    const autohideHandle = setTimeout(() => {
-      onCloseRef.current();
-    }, delayRef.current);
-
-    return () => {
-      clearTimeout(autohideHandle);
+const Toast = React.forwardRef(
+  (
+    {
+      bsPrefix,
+      className,
+      children,
+      transition: Transition,
+      show,
+      animation,
+      delay,
+      autohide,
+      onClose,
+      ...props
+    },
+    ref,
+  ) => {
+    bsPrefix = useBootstrapPrefix('toast');
+    const delayRef = useRef(delay);
+    const onCloseRef = useRef(onClose);
+
+    // We use refs for these, because we don't want to restart the autohide
+    // timer in case these values change.
+    delayRef.current = delay;
+    onCloseRef.current = onClose;
+
+    useEffect(() => {
```

Comment:
lets use `@restart/hooks/useTimeout` maybe

Reply:
Fixed in c3fab88.